packages/openapi-v3/src/decorators/request-body.spike.decorator.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright IBM Corp. 2018,2019. All Rights Reserved.
+// Node module: @loopback/openapi-v3
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import { MetadataInspector, ParameterDecoratorFactory } from '@loopback/context';
+import { JsonSchemaOptions } from '@loopback/repository-json-schema';
+import * as _ from 'lodash';
+import { inspect } from 'util';
+import { TS_TYPE_KEY } from '../controller-spec';
+import { resolveSchema } from '../generate-schema';
+import { OAI3Keys } from '../keys';
+import { RequestBodyObject, REQUEST_BODY_INDEX, SchemaObject } from '../types';
+
+const debug = require('debug')('loopback:openapi3:metadata:requestbody');
+
+export type SchemaOptions = JsonSchemaOptions & {
+  // To support the current flavor using 'x-ts-type',
+  // preserve backward compatibility
+  [TS_TYPE_KEY]?: Function,
+  isVisited?: boolean,
+}
+
+function isRequestBodySpec(spec: Partial<RequestBodyObject> | SchemaOptions): spec is Partial<RequestBodyObject> {
+  return (spec as Partial<RequestBodyObject>).description !== undefined
+    || (spec as Partial<RequestBodyObject>).required !== undefined
+    || (spec as Partial<RequestBodyObject>).content !== undefined;
+}
+
+export function requestBody2(
+  specOrModelOrOptions?: Partial<RequestBodyObject> | Function | SchemaOptions,
+  modelOrOptions?: Function | SchemaOptions,
+  schemaOptions?: SchemaOptions
+) {
```

Comment:
Also to avoid confusion about which parameter takes precedence (`spec.content['application/json'].schema` vs. `model` + `schemaOptions`), I am proposing to accept `model` only when `spec` DOES NOT contain `content` property.

```ts
export function requestBody2(
  spec: Omit<Partial<RequestBodyObject>, 'content'>,
  model: Function,
  schemaOptions?: SchemaOptions
);
```

Reply:
@bajtos Typescript doesn't support "different number of parameters",
see https://www.tutorialsteacher.com/typescript/function-overloading

So we still need the wrapper function to resolve the caller's arguments.

Reply:
@jannyHou TypeScript cannot distinguish between `spec` and `options` from function overloading perspective.

Reply:
> Also to avoid confusion about which parameter takes precedence (spec.content['application/json'].schema vs. model + schemaOptions), I am proposing to accept model only when spec DOES NOT contain content property.

+1, let's implement it in the real PR.